source/core/StringUtilities.cpp

```diff
@@ -118,25 +119,41 @@ Boolean SubString::CompareCStr(ConstCStr begin2) const
     // The source string cannot have a null before it hit end
     // So if a null is found in the supplied string first the test bails out false.
     while(sCompare < sEnd) {
-        if (*sCompare++ != *pBegin2++) {
+        if (*sCompare++ != *pBegin2++)
             return false;
-        }
     }
     // Comparison is true if supplied string is the same length.
     return (*pBegin2 == 0);
 }
 //------------------------------------------------------------
 Boolean SubString::ComparePrefix(const Utf8Char* begin2, Int32 length2) const
 {
-    if (length2 > Length()) {
+    if (length2 > Length())
```

Comment:
We could; I didn't because of performance reasons.  Perhaps we could do it and then mark the shared function inline.

ComparePrefix needs a boolean argument for ignoreCase because is some cases the ignoreCase option is programmatic and not known at compile time.  This is even called directly from some prims.
I wanted a version in which ignoreCase IS known at compile time for performance, since this function is also very commonly called during parsing.
ComparePrefix is only used during parsing and we don't need a programmatic ignoreCase argument because there would be no callers.